File: src/main/java/org/gradle/builds/assemblers/ProjectInitializer.java
package org.gradle.builds.assemblers;

import org.gradle.builds.model.Project;

/**
 * Responsible for setting the initial state for a project.
 */
public abstract class ProjectInitializer {
    public abstract void initRootProject(Project project);

    public abstract void initLibraryProject(Project project);

    public void initAlternateLibraryProject(Project project) {
        initLibraryProject(project);
    }
}

File: src/main/kotlin/org/gradle/builds/assemblers/ProjectInitializer.kt
package org.gradle.builds.assemblers

import org.gradle.builds.model.Project

abstract class ProjectInitializer {
    abstract fun initRootProject(project: Project)

    abstract fun initLibraryProject(project: Project)

    open fun initAlternateLibraryProject(project: Project) {
        initLibraryProject(project)
    }
}